File: app/src/main/java/in/encrust/writersden/ImageAdapter.java
package in.encrust.writersden;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

public class ImageAdapter extends RecyclerView.Adapter<ImageAdapter.ViewHolder> {
    private Context context;
    private List<Drawable> imageList;

    public ImageAdapter(Context context, List<Drawable> imageList) {
        this.context = context;
        this.imageList = imageList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.images, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, final int i) {
        final Drawable image = imageList.get(i);
        viewHolder.imageView.setImageDrawable(image);

        viewHolder.imageView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                if (context instanceof Home){
                    if (i==0){
                        ((Home) context).openGallery();
                    }else {
                        ((Home) context).selectImage(i);
                    }
                }
            }
        });

    }

    @Override
    public int getItemCount() {
        return imageList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ImageView imageView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            imageView = itemView.findViewById(R.id.images);
        }
    }
}

File: app/src/main/java/in/encrust/writersden/ImageAdapter.kt
package `in`.encrust.writersden

import android.content.Context
import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import java.util.ArrayList

class ImageAdapter internal constructor(private val context: Context, private val imageList: ArrayList<Drawable>?) : RecyclerView.Adapter<ImageAdapter.ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.images, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        val image = imageList?.get(i)
        viewHolder.imageView.setImageDrawable(image)

        viewHolder.imageView.setOnClickListener {
            if (context is Home) {
                if (i == 0) {
                    context.openGallery()
                } else {
                    context.selectImage(i)
                }
            }
        }

    }

    override fun getItemCount(): Int {
        return imageList!!.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var imageView: ImageView = itemView.findViewById(R.id.images)

    }
}
